react-native-git-upgrade/cliEntry.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import shell from 'shelljs';
+import Promise from 'promise';
+import yeoman from 'yeoman-environment';
+import TerminalAdapter from 'yeoman-environment/lib/adapter'
+import log from 'npmlog';
+import rimraf from 'rimraf';
+
+import {
+  checkDeclaredVersion,
+  checkMatchingVersions,
+  checkReactPeerDependency,
+  checkGitAvailable,
+  checkNewVersion
+} from './checks';
+
+log.heading = 'git-upgrade';
+
+/**
+ * Promisify the callback-based shelljs function exec
+ * @param command
+ * @param opts
+ * @returns {Promise}
+ */
+function exec(context, command, forceOutput) {
+  return new Promise((resolve, reject) => {
+    shell.exec(command, {silent: (forceOutput ? !forceOutput : !context.cliArgs.verbose)}, (code, stdout, stderr) => {
+      code
+        ? reject(new Error(`Command '${command}' exited with code ${code}:
+stderr: ${stderr}
+stdout: ${stdout}`))
+        : resolve(stdout);
+    });
+  });
+}
+
+function readProjectDescriptors() {
+  try {
+    const rnPakPath = path.resolve(
+      process.cwd(),
+      'node_modules',
+      'react-native',
+      'package.json'
+    );
+
+    const pakPath = path.resolve(
+      process.cwd(),
+      'package.json'
+    );
+
+    const rnPak = JSON.parse(fs.readFileSync(rnPakPath, 'utf8'));
+    const pak = JSON.parse(fs.readFileSync(pakPath, 'utf8'));
+
+    return {rnPak, pak};
+  } catch (err) {
+    throw new Error(
+      'Unable to find project descriptors. Make sure that you have run `npm install` ' +
```

Comment:
What are descriptors? Is this standard npm terminology for package.json files?

How about the following? `Unable to find files ${rnPakPath} and ${pakPath}. Make sure you ran ...`